src/dex/balancer-v3/getPoolsApi.ts

```diff
@@ -69,37 +72,53 @@ function createQuery(
             address
           }
         }
+        hook {
+          address
+        }
       }
     }
   `;
 }
 
-function toImmutablePoolStateMap(pools: Pool[]): ImmutablePoolStateMap {
-  return pools.reduce((map, pool) => {
-    const immutablePoolState: CommonImmutablePoolState = {
-      poolAddress: pool.id,
-      tokens: pool.poolTokens.map(t => t.address),
-      tokensUnderlying: pool.poolTokens.map(t =>
-        t.underlyingToken ? t.underlyingToken.address : null,
-      ),
-      weights: pool.poolTokens.map(t =>
-        t.weight ? parseUnits(t.weight, 18).toBigInt() : 0n,
-      ),
-      poolType: pool.type,
-      // TODO add scalingFactors once API provides them
-      // scalingFactors: pool.poolTokens.map(t => parseUnits('1', 18).toBigInt()),
-      // TODO Hook support will be added in future PR
-      hookType: undefined,
-    };
-
-    map[pool.id] = immutablePoolState;
-    return map;
-  }, {} as ImmutablePoolStateMap);
+function toImmutablePoolStateMap(
+  pools: Pool[],
+  hooksConfigMap: HooksConfigMap,
+): ImmutablePoolStateMap {
+  return (
+    pools
+      // First filter out pools with hooks that aren't in hooksMap
+      .filter(
+        pool =>
+          !pool.hook ||
+          (pool.hook && pool.hook.address.toLowerCase() in hooksConfigMap),
+      )
+      .reduce((map, pool) => {
+        const immutablePoolState: CommonImmutablePoolState = {
+          poolAddress: pool.id,
+          tokens: pool.poolTokens.map(t => t.address),
+          tokensUnderlying: pool.poolTokens.map(t =>
+            t.underlyingToken ? t.underlyingToken.address : null,
+          ),
+          weights: pool.poolTokens.map(t =>
+            t.weight ? parseUnits(t.weight, 18).toBigInt() : 0n,
+          ),
+          poolType: pool.type,
+          hookAddress: pool.hook ? pool.hook.address.toLowerCase() : undefined,
+          hookType: pool.hook
+            ? hooksConfigMap[pool.hook.address.toLowerCase()].type
+            : undefined,
+          supportsUnbalancedLiquidity: true,
```

Comment:
Is it `true` for all supported pools now?

Reply:
Its ok to default to true here as this is only required for balancer maths add/remove which we don't use.